picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,168 @@
+        "box next to the settings, and then enable the removal by checking the \"Remove "
+        "selected options\" box. When you choose \"Make It So!\" to save your option "
+        "settings, the selected items will be removed."
+    )
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.ui = Ui_MaintenanceOptionsPage()
+        self.ui.setupUi(self)
+        self.ui.description.setText(_(self.INSTRUCTIONS))
+        self.ui.tableWidget.setHorizontalHeaderLabels([_("Option"), _("Value")])
+
+    def load(self):
+        from picard.plugin import _extension_points
+        config = get_config()
+
+        # Setting options from all option pages and loaded plugins (including plugins currently disabled).
+        key_options = set(config.setting.as_dict().keys())
+
+        # All setting options included in the INI file.
+        file_options = set([key[8:] for key in config.allKeys() if key[:8] == 'setting/'])
+
+        # Combine all page and plugin settings with required options not appearing in option pages.
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        def _add_options_from_page(page, current_options):
+            if not hasattr(page, 'options'):
+                return
+            for option in page.options:
+                if option.section == "setting":
+                    current_options.add(option.name)
+
+        for ep in _extension_points:
+            if ep.label == 'pages':
+                for page in ep:
+                    _add_options_from_page(page, current_options)
+
+        orphan_options = file_options.difference(current_options)
+
+        self.ui.description.setText(
+            _(self.INSTRUCTIONS)
+            + _("\n\nThe configuration file currently contains {0} option settings, {1} which are unused.").format(
+                len(file_options),
+                len(orphan_options)
+            )
+        )
+        self.ui.enable_cleanup.setChecked(False)
+        self.ui.tableWidget.clearContents()
+        self.ui.tableWidget.setRowCount(len(orphan_options))
+        for row, item in enumerate(sorted(orphan_options)):
+            tableitem = QtWidgets.QTableWidgetItem(item)
+            tableitem.setFlags(tableitem.flags() | QtCore.Qt.ItemIsUserCheckable)
+            tableitem.setCheckState(False)
+            self.ui.tableWidget.setItem(row, 0, tableitem)
+            text = self.make_setting_value_text(item)
+            line_height = max(25, 18 * min(5, text.count("\n") + 1))
```

Comment:
Why this calculation? Please explain.

Reply:
This is to expand the row height to a reasonable size if there are multiple lines in a "Value".  The default size is only one line with tiny scroll bar.  Should I remove this and just leave the default line height, or leave it in and add a comment with an explanation?

Reply:
Comment is fine, thanks.